src/App.js

```diff
@@ -1,85 +1,257 @@
+    localStorage.setItem('list', JSON.stringify(this.state));
+  }
+
+  handleSubmit = (event, newTodoText) => {
+    event.preventDefault();
+
+    if (newTodoText.replace(/ /g, '') === '') {
+      return;
+    }
+
+    const sample = {
+      id: this.state.id + 1,
+      text: newTodoText.trim(),
+      elementState: {
+        completed: false,
+        editing: false,
+      },
+    };
+
+    this.setState(state => ({
+      ListofTodo: state.ListofTodo.concat([{ ...sample }]),
+      id: state.id + 1,
+    }));
+  }
+
+  handleCheckboxChange = (id) => {
+    this.setState(state => ({ ListofTodo: state.ListofTodo.map((todo) => {
+      if (todo.id !== id) {
+        return todo;
+      }
+
+      return {
+        ...todo,
+        elementState: {
+          ...todo.elementState,
+          completed: !todo.elementState.completed,
+        },
+      };
+    }) }));
+  }
+
+  handleClickDestroy = (id) => {
+    this.setState(state => ({ ListofTodo: state.ListofTodo
+      .filter(todo => todo.id !== id) }));
+  }
+
+  handleDoubleClick = (id, input) => {
+    this.setState(state => ({
+      ListofTodo: state.ListofTodo.map((todo) => {
+        if (todo.id !== id) {
+          return todo;
+        }
+
+        return {
+          ...todo,
+          elementState: {
+            ...todo.elementState,
+            editing: true,
+          },
+        };
+      }),
+    }));
+  }
+
+  handleEditing = (event, id, prevValue) => {
+    const { value } = event.target;
+
+    if (event.key === 'Enter') {
+      this.setState(state => ({ ListofTodo: state.ListofTodo.map((todo) => {
+        if (todo.id !== id) {
+          return todo;
+        }
+
+        return {
+          ...todo,
+          text: value,
+          elementState: {
+            ...todo.elementState,
+            editing: false,
+          },
+        };
+      }) }));
+    }
+
+    if (event.key === 'Escape') {
+      this.setState(state => ({ ListofTodo: state.ListofTodo.map((todo) => {
+        if (todo.id !== id) {
+          return todo;
+        }
+
+        return {
+          ...todo,
+          text: prevValue,
+          elementState: {
+            ...todo.elementState,
+            editing: false,
+          },
+        };
+      }) }));
+    }
+  }
+
+  handleLossFocus = (event, id) => {
+    const { value } = event.target;
+
+    this.setState(state => ({ ListofTodo: state.ListofTodo.map((todo) => {
+      if (todo.id !== id) {
+        return todo;
+      }
+
+      return {
+        ...todo,
+        text: value,
+        elementState: {
+          ...todo.elementState,
+          editing: false,
+        },
+      };
+    }) }));
+  }
+
+  handleToggleAll = (event) => {
+    const { checked } = event.target;
+
+    if (checked) {
+      this.setState(state => ({ ListofTodo: state.ListofTodo.map(todo => ({
+        ...todo,
+        elementState: {
+          ...todo.elementState,
+          completed: true,
+        },
+      })) }));
```

Comment:
```suggestion
      this.setState(state => ({ ListofTodo: state.ListofTodo.map(todo => ({
        ...todo,
        elementState: {
          ...todo.elementState,
          completed: checked,
        },
      })) }));
```
one action is enough, there is no need to make a condition wrapper